pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -354,6 +390,24 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 	return dataFromTemplate, nil
 }
 
+func convertRegistryMapToSliceInOrder(registryMap map[int]map[string]interface{}) []map[string]interface{} {
```

Comment:
This function is not robust enough to handle the example we spoke about "offline." It will panic for this example you provided:
>let’s say len(existingRegistries) = 0 and the data in the api call is
"rancherKubernetesEngineConfig.privateRegistries[3].url":"aaa",
"rancherKubernetesEngineConfig.privateRegistries[3].user":"aaa"
"rancherKubernetesEngineConfig.privateRegistries[3].password":"aaa"

This is because `len(registryMap) == 1`, so the call to `registries[k] = v` will be out of bounds because `k == 3`.

I think handling this properly is equivalent to not using the `map[int]map[string]interface{}` at all. Meaning, that if you can solve this problem, then you would have to use the `map[int]map[string]interface{}` and would be able to continue to use the `[]map[string]interface{}` above.

Reply:
discussed offline and this is properly handled now.